src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,44 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate fromDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate toDate = LocalDate.parse(dateTo, formatter);
+
+        int[] totalSalary = new int[names.length];
+        StringBuilder result = new StringBuilder();
+        result.append("Report for period ")
+            .append(dateFrom).append(" - ").append(dateTo).append("\n");
+
+        for (String line : data) {
+            String[] parts = line.split(" ");
+            LocalDate date = LocalDate.parse(parts[0], formatter);
+            String name = parts[1];
+            int hours = Integer.parseInt(parts[2]);
+            int rate = Integer.parseInt(parts[3]);
```

Comment:
All magic numbers, strings, symbols in your code should be constants.
[this article](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)